src/main/java/gov/cabinetoffice/api/mappers/CustomSubmissionMapperImpl.java

```diff
@@ -79,22 +77,63 @@ public AddressDTO buildAddress(String[] multiResponse) {
 		return SubmissionMapper.super.buildAddress(multiResponse);
 	}
 
+	// TODO - look into whether this is horrendously inefficient. Potentially more performant to do all this looping at query level.
```

Comment:
May only be able to test this with large quantities of data but I'll have a go over the coming days.

Reply:
Yeah thinking we may want to refactor this and do more at the database level. If we're pulling _all_ submissions then performing filtering in-memory that could get very slow.